flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/SlotSharingExecutionSlotAllocator.java

```diff
@@ -213,20 +235,24 @@ private ResourceProfile getPhysicalSlotResourceProfile(ExecutionSlotSharingGroup
 			.reduce(ResourceProfile.ZERO, (r, e) -> r.merge(resourceProfileRetriever.apply(e)), ResourceProfile::merge);
 	}
 
-	private SharingPhysicalSlotRequestBulk createBulk(Map<ExecutionSlotSharingGroup, List<ExecutionVertexID>> executions) {
-		Map<ExecutionSlotSharingGroup, ResourceProfile> pendingRequests = executions
-			.keySet()
-			.stream()
-			.collect(Collectors.toMap(
-				group -> group,
-				group -> sharedSlots.get(group).getPhysicalSlotResourceProfile()
-			));
+	private Optional<SharingPhysicalSlotRequestBulk> createBulk(
+			Map<ExecutionSlotSharingGroup, List<ExecutionVertexID>> executions) {
+		Map<ExecutionSlotSharingGroup, ResourceProfile> pendingRequests = new HashMap<>();
+		for (ExecutionSlotSharingGroup group : executions.keySet()) {
+			SharedSlot sharedSlot = sharedSlots.get(group);
+			if (sharedSlot == null || sharedSlot.getSlotContextFuture().isCompletedExceptionally()) {
+				// there is no shared slot for this group or its physical slot has already failed
+				// hence there is no point to track the whole bulk
+				return Optional.empty();
+			}
```

Comment:
Here it is a bit the same with the special case handling. What exactly makes the cases 1) directly failed physical slot future and 2) failing just after this method completes different?